src/hooks/users.ts

```diff
@@ -79,81 +117,68 @@ export const useRecentWorldsUsers = (): {
   return useMemo(
     () => ({
       recentWorldUsers: worldUsers.filter(
-        (user) => user.lastSeenAt > lastSeenThreshold
+        (user) =>
+          normalizeTimestampToMilliseconds(user.lastSeenAt) > lastSeenThreshold
       ),
       isRecentWorldUsersLoaded: isWorldUsersLoaded,
     }),
     [worldUsers, isWorldUsersLoaded, lastSeenThreshold]
   );
 };
 
-export const useRecentVenueUsers = (): {
-  recentVenueUsers: readonly WithId<User>[];
-  isRecentVenueUsersLoaded: boolean;
-} => {
+export const useRecentLocationUsers = (locationName?: string) => {
```

Comment:
I don't think this function should know/care about the number of possible location names. Its sole purpose is to filter users based on the location, which aligns with the Single responsibility principle, I suppose. I feel strongly about not changing the way it is if no serious arguments are presented.

Added the documentation comment

Reply:
For some reason I can't comment under https://github.com/sparkletown/sparkle/pull/1047#discussion_r565025279. So will answer here.

Even though I do understand your logic on this, are there any obvious cases, when this can be beneficial? I see it as overengineering without any real value. I think we should also keep in mind that this is a workaround, not a complete solution.

Reply:
Answer for https://github.com/sparkletown/sparkle/pull/1047#discussion_r565024671

1)I think we had a very similar discussion on `useCurrentVenueNG`'s params.
2) Is it possible to have 2 different values returned from `useVenueId` simultaneously in different parts of the app?

Reply:
RE: https://github.com/sparkletown/sparkle/pull/1047#discussion_r565205669

> I don't think this function should know/care about the number of possible location names. Its sole purpose is to filter users based on the location, which aligns with the Single responsibility principle, I suppose. I feel strongly about not changing the way it is if no serious arguments are presented.

My desire here is that we don't have the logic for constructing what a 'valid location' looks like spread across the codebase like it has been in the past. There should be a single canonical thing used, that combines the 'base objects' to create a valid location.

That doesn't need to be this function directly though; perhaps there are some helper functions that 'construct from base primitives (eg. `venueId`, `roomName`) that return a specially typed string (eg. `Location`) that this function accepts as it's parameter. That way we have the 'single responsibility principle', enforced type safety (rather than 'magic strings'), and centralise the logic for constructing valid locations from 'base objects'.

---

RE: https://github.com/sparkletown/sparkle/pull/1047#discussion_r565208747

> Even though I do understand your logic on this, are there any obvious cases, when this can be beneficial? I see it as overengineering without any real value. I think we should also keep in mind that this is a workaround, not a complete solution.

I believe I already stated the case in my previous review comments: https://github.com/sparkletown/sparkle/pull/1047#discussion_r565024671

> Alternatively, if this is designed only to work with the current venue (since it's overwriting a key in a global store, without namespacing it by the venueId we are attempting to lookup here) perhaps we don't need to pass it in at all (not sure that is my preference, but mentioning it here as one alternative)
> 
> Basically, we need to ensure that this works even if different `venueIds` are passed in in different areas of the app. As currently written, it has the same basic 'conflicting global store' issues as using `react-redux-firebase` using the same `storeAs` key does.

--- 

RE: https://github.com/sparkletown/sparkle/pull/1047#discussion_r565211713

> I think we had a very similar discussion on `useConnectCurrentVenueNG`'s params. https://github.com/sparkletown/sparkle/pull/1047#discussion_r562364242

Yup, fair enough. Though I think this may also potentially play into my 'support multiple venue ids' comments referenced above/originally in https://github.com/sparkletown/sparkle/pull/1047#discussion_r565024671

Specifically:

> Alternatively, if this is designed only to work with the current venue (since it's overwriting a key in a global store, without namespacing it by the `venueId` we are attempting to lookup here) perhaps we don't need to pass it in at all (not sure that is my preference, but mentioning it here as one alternative)

---

> Is it possible to have 2 different values returned from `useVenueId` simultaneously in different parts of the app?

I would expect probably not (or if it can be, it would be a race condition). But my comments there were more related to the fact that you are allowing an arbitrary `venueId` to be passed into it.

While your intent/current usage may be that that `venueId` will only ever come from `useVenueId`, that's not something that can be guaranteed when it's a 'free param' for me to enter. There is a very high likelihood that someone (eg. myself) may come along and be like "cool, a hook that lets me get the 'sovereign venue id' based on the `venueId` I pass to it!" and then pass it a `venueId` that differs from what `useVenueId` would return, which would then break things.

When using global state, we need to design/think about it in a way that we avoid the potential of these issues, particularly when it doesn't actually take a huge amount of extra code to build it 'defensively' up front.